src/services/review/RepoCheckerService.ts

```diff
@@ -77,33 +78,57 @@ export default class RepoCheckerService {
     this.pageService = pageService
   }
 
-  isCurrentlyLocked(repo: string): ResultAsync<true, SiteCheckerError> {
+  createLogsFolder(
+    repo: string
+  ): ResultAsync<string | undefined, SiteCheckerError> {
     const logsFilePath = path.join(SITE_CHECKER_REPO_PATH, repo)
     // create logs folder if it does not exist
+    return ResultAsync.fromPromise(doesDirectoryExist(logsFilePath), (err) => {
+      logger.error(
+        `SiteCheckerError: error finding if logs directory exists for ${repo} :${err}`
+      )
+      return new SiteCheckerError(
+        `Error finding if logs directory exists for ${repo} :${err}`
+      )
+    }).andThen((doesLogsFolderExist) => {
+      if (!doesLogsFolderExist) {
+        return ResultAsync.fromPromise(
+          fs.promises.mkdir(logsFilePath, { recursive: true }),
+          (err) => {
+            logger.error(`SiteCheckerError: Unable to create logs dir: ${err}`)
+            return new SiteCheckerError(
+              `SiteCheckerError: Unable to create logs dir: ${err}`
+            )
+          }
+        )
+      }
+      return okAsync("ok")
+    })
+  }
 
-    if (!fs.existsSync(logsFilePath)) {
-      fs.mkdirSync(logsFilePath, { recursive: true })
-    }
-
+  isCurrentlyLocked(repo: string): ResultAsync<true, SiteCheckerError> {
+    const logsFilePath = path.join(SITE_CHECKER_REPO_PATH, repo)
     // check if checker.lock exists
     const lockFilePath = path.join(logsFilePath, LOCK_CONTENT)
-
-    return ResultAsync.fromPromise(
-      fs.promises.access(lockFilePath, fs.constants.F_OK),
-      (err) => new SiteCheckerError(`${err}`)
-    ).map(() => true)
+    return this.createLogsFolder(repo)
+      .andThen(() =>
+        ResultAsync.fromPromise(
+          fs.promises.access(lockFilePath, fs.constants.F_OK),
+          (err) => new SiteCheckerError(`${err}`)
```

Comment:
we don't have to fix this now but note that we can't differentiate between `SiteCheckerError` and (an imaginary) `FolderLockError`